headers/augment_headers.py

```diff
@@ -289,6 +293,13 @@ def add_header_content(
         action="store_true",
         help="add alias declarations",
     )
+    parser.add_argument(
+        "-ad",
+        "--mark_aliases_deprecated",
```

Comment:
This name seems a bit long. Also I usually like CLI flags to be dash-separated, not underscore-separated. How about `--deprecate-aliases`?